### PR TITLE
[synopsys-ptpx-genlibdb] - Update script to check if spef files exist

### PR DESCRIPTION
**Pull request by Gu3nol**
This step can now be used after synthesis to generate lib files
The spef files are no more mandatory as inputs of the step

### CODE REVIEW

File: steps/synopsys-ptpx-genlibdb/configure.yml [hunk 1]
```diff
@@ -49,7 +49,6 @@ preconditions:
   - assert Tool( 'pt_shell' )                # tool check
   - assert File( 'inputs/adk' )              # must exist
   - assert File( 'inputs/design.vcs.v' )     # must exist
-  - assert File( 'inputs/design.spef.gz' )   # must exist
```
Comment (ctorng) [inline steps/synopsys-ptpx-genlibdb/configure.yml]: Can you actually leave this in (but commented out), and then replace "must exist" with "spef is optional"?

That way people don't think we just forgot and then decide to add it back in.

File: steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl [hunk 2]
```diff
@@ -28,6 +28,18 @@ current_design $ptpx_design_name
 link_design
 
 # Read in the SDC and parasitics
-
-read_sdc -echo $ptpx_sdc
-read_parasitics -format spef $ptpx_spef
+# Try to read the sdc constraints files
+if {[ file exists $ptpx_sdc ]} {
```
Comment (ctorng) [inline steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl]: Super minor, but for coding convention can we add an empty line between the comment and the if statement?

File: steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl [hunk 3]
```diff
@@ -28,6 +28,18 @@ current_design $ptpx_design_name
...
+    puts "\n  > Info: Sourcing $ptpx_sdc\"\n"
+    read_sdc -echo $ptpx_sdc
+}  else {
+    puts "\n  > Warn: No sdc constraint file found\"\n"
+}
+
+# Try to read the sped parasitic files
```
Comment (ctorng) [inline steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl]: Same thing here with an empty line.

Also "sped" -> "spef"

File: steps/synopsys-ptpx-genlibdb/configure.yml (hunk 1 above)
Comment (Gu3nol) [reply steps/synopsys-ptpx-genlibdb/configure.yml]: Done. Actually DC is generating a spef file in the report.
I added some step to compress this file in .gz form to be used

File: steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl (hunk 2 above)
Comment (Gu3nol) [reply steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl]: Done. :)

File: steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl (hunk 3 above)
Comment (Gu3nol) [reply steps/synopsys-ptpx-genlibdb/scripts/read_design.tcl]: Done. :)

**Comment by Gu3nol** [comment]
Fix Minor issues
Add script to compress the spef file generated by dc in synthesis to be used as an input of the genlib step

**Pull request merged**